meilisearch-http/src/routes/indexes/documents.rs

```diff
@@ -86,14 +83,31 @@ pub fn configure(cfg: &mut web::ServiceConfig) {
     );
 }
 
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct GetDocument {
+    fields: Option<String>,
+}
+
 pub async fn get_document(
     meilisearch: GuardedData<ActionPolicy<{ actions::DOCUMENTS_GET }>, MeiliSearch>,
     path: web::Path<DocumentParam>,
+    params: web::Query<GetDocument>,
 ) -> Result<HttpResponse, ResponseError> {
     let index = path.index_uid.clone();
     let id = path.document_id.clone();
+    let attributes_to_retrieve = params.fields.as_ref().and_then(|attrs| {
+        let mut names = Vec::new();
+        for name in attrs.split(',').map(String::from) {
+            if name == "*" {
+                return None;
+            }
+            names.push(name);
+        }
+        Some(names)
+    });
```

Comment:
@Kerollmops introduced `StarOr` in #2399, we might want to use It here, since it was merged

Reply:
Oooh yeah, `StarOr` is pretty cool. I moved it to `meilisearch-http/src/routes/mod.rs` since it's a really front-end thing I think :thinking: 
I also took the opportunity to simplify it a bit, instead of writing the `fold` ourselves I use the std implementation of `collect` to transform an `Iterator<Item = Option<T>>` into an `Option<Vec<T>>`.